cmake/CompilerHelper.cmake

```diff
@@ -3,4 +3,8 @@
 # =============================================================================
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
+endif()
+
+if (CMAKE_C_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS} -undefined dynamic_lookup")
```

Comment:
Shouldn't this be activated purely for Python modules? Or all libraries are expected to be not-self-contained?
I also find `list(APPEND ...)` nicer

Reply:
> Shouldn't this be activated purely for Python modules
it's for other libraries as well...for example, if libnrnmpi.so needs some symbols from libnrniv.so, on OSX we get undefined symbol.

> I also find list(APPEND ...) nicer

but `CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS` is string and not LIST.